Handle 400-type errors with BadRequestError plus small unit test fix

mpistrang opened the pull request:
This fixes the TypeError raised by not passing enough arguments to cache.set().

Commit:
pass key and value to cache.set()

mpistrang commented:
@jacobb please check out this small change to one of the unit tests within Nap.  Once fixed, all the tests will pass!

Thanks,
-Mike

Commit:
add bad_request_status to the DFEAULT_CONFIG

Commit:
check for bad request on updates and creates

Commit:
fix test coverage of BadStatusError

Commit:
BadRequestError should be a subclass of InvalidStatusError

Commit:
pass the expected status to the exception for more useful information

Commit:
update BadRequestError to include errors as exception msg

Commit:
update version

mcordes committed:
ensure there is a slash between the base url and resource uri fragments

mcordes committed:
Revert "ensure there is a slash between the base url and resource uri fragments"

This reverts commit 5e089909ee5e8a6d540547cd4f5e9fbbbb55145b.

mcordes committed:
fix lame handling of slashes when joining base url and resource uri

Commit:
update version for trailing slash handling

Commit:
Merge branch 'better_root_url_trailing_slash_handling'

mcordes committed:
better error handling of cache header logic

mcordes committed:
test non-max age cache control value

mcordes committed:
Merge pull request #6 from CashStar/better_cache_control_error_handling

better error handling of cache header logic

mcordes committed:
increase version#

ccooper123456 committed:
INFR-930 changing nap to always generate a version based on date time installed not version number.  if this was package based the number would be better, having source controlled based means changes are not installed but are still downloaded causing confusion between developers.

mcordes committed:
add new fields: decimal field, simple list field

mcordes committed:
add new nap fields

mcordes committed:
handle tuples, add tests, verify via test that SimpleList likes falsy values

mcordes committed:
Merge pull request #7 from CashStar/add_new_nap_fields

Add new nap fields

mcordes committed:
2to3

mcordes committed:
py3 - bytes vs strings

mcordes committed:
branch compatible with python2 or 3

mcordes committed:
better handling of response.content which is a byte array in py3 and a unicode string in py2

mcordes committed:
simplify for py2 and py3

mcordes committed:
handle Nones in to_unicode

mcordes committed:
pr feedback - remove extraneous list(...); add absolute imports

mcordes committed:
dont return None from unicode (its exposing a bug in econsumer Im not sure were ready to fix yet; and the old version didnt return None here)

mcordes committed:
fix to_unicode

Commit:
remove the .idea (IntelliJ) directory from git

Commit:
add a newline to the end of the .gitignore file

Commit:
ILL-1487 add url normalization to prevent invalid caching

This commit adds a utility function, normalize_url, that is
used in _generate_url to sort query parameters in a url
while maintaining the order of lists and duplicate keys.

The purpose of this change is to prevent the situation
where extra parameters are used in a resource lookup,
but may be passed in different ways, causing the same entity
to be cached using more than one url leading
to invalid cache states and cache misses.

Commit:
ILL-1487 remove unintended whitespace added

Commit:
ILL-1487 update lookup url test to try all dict orderings

Commit:
ILL-1487 update test to use OrderedDict in place of Dict

This is done to ensure we actually try all the different permutations.

dcampbell24 committed:
Merge pull request #13 from CashStar/ILL-1487

ILL-1487 add url normalization to prevent invalid caching

7/12 thumbs up, no unanswered questions, no thumbsdown

jacobb commented:
@mpistrang @dcampbell24 This has gotten fairly large for a single PR, but I'd be interested in getting hte branches caught up. Any chance we could parse things out a bit?

mcordes committed:
fix some imports so it works with py3 agagain

dcampbell24 commented:
@jacobb would squashing the commits be helpful? We would also like to see the branches caught up, but a few weeks may pass before anyone has enough time to clean-up the PR or create a new one.

jugallo committed:
encode unicode strings to UTF-8 so that python can urlencode them

jugallo committed:
Adding back the normalize function and checking for six.text_type instead of unicode

jugallo committed:
revmoving normalize_url again

jugallo committed:
remove test cases for normalize_url (which I've deleted)

jugallo committed:
move a comment to the relevant function and add a test for unicode url parameters

jugallo committed:
make safe_encode a top-level function

jugallo committed:
replace some tests that I removed previously

jugallo committed:
use itemgetter instead of a lambda function as the key in my sorted call.

jugallo committed:
Merge pull request #15 from CashStar/NI-3570-unicode-error

encode unicode strings to UTF-8 so that python can urlencode them

mcordes committed:
ARCH-1867 - fix handling of unicode characters in resource repr

Commit:
Update DecimalField to handle empty and 0 values when scrubbing and descrubbing

Commit:
update docstring to indicate cast-to-string before constructing decimal

mpistrang committed:
Merge pull request #17 from CashStar/decimal_updates

Update DecimalField to distinguish between empty and 0 values

matteius committed:
Add concept of hashing a cache key above a specified length

This allows us to work around memcached limits while keeping, mostly,
plaintext keys in memcached.

matteius committed:
Fix test to work with python 3.5

Maintains compatibility with older and newer python versions.

matteius committed:
Add a unit test that a long cache key is hashed

matteius committed:
Add logging when cache key size is exceeded

Log the original key and the hashed key.

matteius committed:
change default nap log level to INFO

matteius committed:
Merge pull request #19 from CashStar/cache-key-length

Support md5 hashing part of the cache key above a certain length

jkomusin committed:
Add caching backend for Flask-Caching (#18)

* Added Flask-Caching support and refactored testing harness and requirements

nap will no longer install 'mock' as a required dependency, and lists all dependencies required for its full suite of tests within setuptools' test requirements. Now 'make test' will ensure that all test dependencies are available before running the test suite, rather than skipping framework-specific tests.

* Refactored cache tests

Minimized the scope of mocks and removed pytest importorskip now that we're guaranteed to have all of the frameworks for our tests.

jdsmith committed:
Updated unicode encoding to fix docker pip install issue.

jdsmith committed:
Updated utf to utf-8

jdsmith committed:
Used io.open instead of open to maintain python 2 compatibility.

Commit:
Merge pull request #21 from CashStar/docker_ascii_fix

Updated unicode encoding to fix docker pip install issue.

mcordes committed:
merge from ARCH-2849_better_handling_of_bad_cache_data

Commit:
ARCH-3001 Make log message more similar to cstar_clients and remove the trying to hit language

matteius committed:
Merge pull request #23 from CashStar/ARCH-3001

ARCH-3001 Improve logging in nap

mcordes committed:
NI-5221 - add caching of GET collection lookup

mcordes committed:
make it more clear were caching the response

mcordes committed:
add tests

mcordes committed:
remove extra space

mcordes committed:
update comment

mcordes committed:
test cleanup

mcordes committed:
Merge pull request #24 from CashStar/NI-5221_clients_performance_issue

Ni 5221 clients performance issue

sprytnyk committed:
📝 ARCH-3394 Update examples, fix PEP 8 issues, align it according to Python 3.

sprytnyk committed:
🔧 ARCH-3394 Drop installation of six & mock, put a Python 3 label.

sprytnyk committed:
✨ ARCH-3394 Migrate codebase to Python 3, implement various fixes of code, drop redundancy, PEP 8, etc.

sprytnyk committed:
✅ ARCH-3394 Migrate tests codebase to Python 3, implement various fixes of code, drop redundancy,  fix PEP 8, etc.

sprytnyk committed:
📝 ARCH-3394 Fix a string.

matteius committed:
Merge pull request #25 from vald-phoenix/feature/ARCH-3394-migrate-nap-to-python-3

Feature/ARCH-3394 Migrate nap codebase to Python 3
